trieste/acquisition/function.py

```diff
@@ -528,15 +532,187 @@ def prepare_acquisition_function(
         return lambda at: expected_improvement(objective_model, eta, at) * constraint_fn(at)
 
 
-class BatchAcquisitionFunctionBuilder(ABC):
-    """ A :class:`BatchAcquisitionFunctionBuilder` builds a batch acquisition function. """
+class IndependentReparametrizationSampler:
+    r"""
+    This sampler employs the *reparameterization trick* to approximate samples from a
+    :class:`ProbabilisticModel`\ 's predictive distribution as
+
+    .. math:: x \mapsto \mu(x) + \epsilon \sigma(x)
+
+    where :math:`\epsilon \sim \mathcal N (0, 1)` is constant for a given sampler, thus ensuring
+    samples form a continuous curve.
+    """
+    def __init__(self, sample_size: int, model: ProbabilisticModel):
+        """
+        :param sample_size: The number of samples to take at each point. Must be positive.
+        :param model: The model to sample from.
+        :raise ValueError (or InvalidArgumentError): If ``sample_size`` is not positive.
+        """
+        tf.debugging.assert_positive(sample_size)
+
+        self._sample_size = sample_size
+        self._eps = tf.Variable(
+            tf.reshape([], [sample_size, 0]), shape=[sample_size, None], dtype=tf.float64
+        )  # [S, 0]
+        self._model = model
+
+    def sample(self, at: QueryPoints) -> tf.Tensor:
+        """
+        Return approximate samples from the `model` specified at :meth:`__init__`. Multiple calls to
+        :meth:`sample`, for any given :class:`IndependentReparametrizationSampler` and ``at``, will
+        produce the exact same samples. Calls to :meth:`sample` on *different*
+        :class:`IndependentReparametrizationSampler` instances will produce different samples.
```

Comment:
@vpicheny you mentioned the implementation wasn't very obvious, and I think you said you'd like me to document how it works. Is this what you meant, or did you mean in finer detail?

Reply:
(see also the class docstring)

Reply:
actually the not obvious bit for me is that self._eps is empty at first and populated the first time sample is called, then it is fixed.

Reply:
it's essentially a lazy constant. shall i add that?